app/models/financialDetails/ChargeItem.scala

```diff
@@ -0,0 +1,240 @@
+                        interestEndDate: Option[LocalDate],
+                        interestRate: Option[BigDecimal],
+                        lpiWithDunningLock: Option[BigDecimal],
+                        amountCodedOut: Option[BigDecimal],
+                        dunningLock: Boolean) extends TransactionItem {
+
+  def isOverdue()(implicit dateService: DateServiceInterface): Boolean =
+    dueDate.exists(_ isBefore dateService.getCurrentDate)
+
+  val hasLpiWithDunningLock: Boolean =
+    lpiWithDunningLock.isDefined && lpiWithDunningLock.getOrElse[BigDecimal](0) > 0
+
+  def hasAccruingInterest: Boolean =
+    interestOutstandingAmount.isDefined && latePaymentInterestAmount.getOrElse[BigDecimal](0) <= 0
+
+  def isAccruingInterest()(implicit dateService: DateServiceInterface): Boolean = {
+    Seq(PaymentOnAccountOneReviewAndReconcile, PaymentOnAccountTwoReviewAndReconcile).contains(transactionType) && !isPaid && !isOverdue()
+  }
+
+  def getDueDateForNonZeroBalancingCharge(codedOutEnabled: Boolean = false): Option[LocalDate] = {
+    if(transactionType == BalancingCharge && (!codedOutEnabled || subTransactionType.isEmpty) && originalAmount == 0.0) {
+      None
+    } else {
+      dueDate
+    }
+  }
+
+  def isPaid: Boolean = outstandingAmount match {
+    case amount if amount == 0 => true
+    case _ => false
+  }
+
+  def isLatePaymentInterest: Boolean = latePaymentInterestAmount match {
+    case Some(amount) if amount <= 0 => false
+    case Some(_) => true
+    case _ => false
+  }
+
+  def isCodingOut: Boolean = {
+    val codingOutSubTypes = Seq(Nics2, Accepted, Cancelled)
+    subTransactionType.exists(subType => codingOutSubTypes.contains(subType))
+  }
+
+  def interestIsPaid: Boolean = interestOutstandingAmount match {
+    case Some(amount) if amount == 0 => true
+    case _ => false
+  }
+
+  def remainingToPay: BigDecimal = {
+    if (isPaid) BigDecimal(0)
+    else outstandingAmount
+  }
+
+  def remainingToPayByChargeOrLpi: BigDecimal = {
+    if (isLatePaymentInterest) interestRemainingToPay
+    else remainingToPay
+  }
+
+  val isPartPaid: Boolean = outstandingAmount != originalAmount
+
+  val interestIsPartPaid: Boolean = interestOutstandingAmount.getOrElse[BigDecimal](0) != latePaymentInterestAmount.getOrElse[BigDecimal](0)
+
+  def getInterestPaidStatus: String = {
+    if (interestIsPaid) "paid"
+    else if (interestIsPartPaid) "part-paid"
+    else "unpaid"
+  }
+
+  def getChargePaidStatus: String = {
+    if (isPaid) "paid"
+    else if (isPartPaid) "part-paid"
+    else "unpaid"
+  }
+
+  def checkIsPaid(isInterestCharge: Boolean): Boolean = {
+    if (isInterestCharge) interestIsPaid
+    else isPaid
+  }
+  def interestRemainingToPay: BigDecimal = {
+    if (interestIsPaid) BigDecimal(0)
+    else interestOutstandingAmount.getOrElse(latePaymentInterestAmount.get)
+  }
+
+  def getChargeTypeKey(codedOutEnabled: Boolean = false, reviewAndReconcileEnabled: Boolean = false): String =
```

Comment:
I think this can be moved into the shared trait for a few models, see similar comments below